modules/dnn/src/layers/scale_layer.cpp

```diff
@@ -291,5 +291,131 @@ Ptr<Layer> ShiftLayer::create(const LayerParams& params)
     return Ptr<ScaleLayer>(new ScaleLayerImpl(scaleParams));
 }
 
+class DataAugmentationLayerImpl CV_FINAL : public DataAugmentationLayer
+{
+public:
+    DataAugmentationLayerImpl(const LayerParams& params)
+    {
+        setParamsFrom(params);
+        recompute_mean = params.get<int>("recompute_mean", 0);
+        mean_per_pixel = params.get<bool>("mean_per_pixel", false);
+        num_iter = 0;
+    }
+
+    bool getMemoryShapes(const std::vector<MatShape> &inputs,
+                         const int requiredOutputs,
+                         std::vector<MatShape> &outputs,
+                         std::vector<MatShape> &internals) const CV_OVERRIDE
+    {
+        CV_Assert(blobs[0].total() == 1);
+        CV_Assert(blobs[1].total() == total(inputs[0], 1));
+        CV_Assert(blobs[2].total() == inputs[0][1]);
+        outputs.assign(1, inputs[0]);
+        return true;
+    }
+
+    virtual void finalize(InputArrayOfArrays inputs_arr, OutputArrayOfArrays) CV_OVERRIDE
+    {
+        std::vector<Mat> inputs;
+        inputs_arr.getMatVector(inputs);
+        CV_Assert(inputs.size() == 1);
+        CV_Assert(!mean_per_pixel && blobs.size() == 3 || blobs.size() >= 2);
+        ++num_iter;
```

Comment:
finalize is called when network is initialized. Shall we do `num_iter = 0` here to reset the state?